homeassistant/components/nextcloud/sensor.py

```diff
@@ -76,6 +268,12 @@ class NextcloudSensor(NextcloudEntity, SensorEntity):
     """Represents a Nextcloud sensor."""
 
     @property
-    def native_value(self) -> StateType:
+    def native_value(self) -> StateType | datetime:
         """Return the state for this sensor."""
-        return self.coordinator.data.get(self.item)
+        val = self.coordinator.data.get(self.item)
+        if (
+            getattr(self.entity_description, "device_class", None)
+            == SensorDeviceClass.TIMESTAMP
+        ):
+            return datetime.fromtimestamp(cast(int, val), tz=UTC)
```

Comment:
I'd add an optional attribute to the entity description with a function that can parse the data before returning the native value. The default can be a no op lambda.

Reply:
good idea :+1 will add this to the above followup PR